components/admin_console/request_button/request_button.tsx

```diff
@@ -16,122 +14,130 @@ import WarningIcon from 'components/widgets/icons/fa_warning_icon';
-         *
-         * If set to true, the action provided in the `saveConfigAction` property will be
-         * called before the action provided in the `requestAction` property, with the later
-         * only being called if the former is successful.
-         */
-        saveNeeded: PropTypes.bool,
-
-        /**
-         * Action to be called to save the config, if saveNeeded is set to true.
-         */
-        saveConfigAction: PropTypes.func,
-
-        /**
-         * True if the success message should be show when the request completes successfully,
-         * otherwise false.
-         */
-        showSuccessMessage: PropTypes.bool,
-
-        /**
-         * The message to show when the request completes successfully.
-         */
-        successMessage: PropTypes.shape({
-
-            /**
-             * The i18n string ID for the success message.
-             */
-            id: PropTypes.string.isRequired,
-
-            /**
-             * The i18n default value for the success message.
-             */
-            defaultMessage: PropTypes.string.isRequired,
-        }),
-
-        /**
-         * The message to show when the request returns an error.
-         */
-        errorMessage: PropTypes.shape({
-
-            /**
-             * The i18n string ID for the error message.
-             */
-            id: PropTypes.string.isRequired,
-
-            /**
-             * The i18n default value for the error message.
-             *
-             * The placeholder {error} may be used to include the error message returned
-             * by the server in response to the failed request.
-             */
-            defaultMessage: PropTypes.string.isRequired,
-        }),
-
-        /**
-         * True if the {error} placeholder for the `errorMessage` property should include both
-         * the `message` and `detailed_error` properties of the error returned from the server,
-         * otherwise false to include only the `message` property.
-         */
-        includeDetailedError: PropTypes.bool,
-
-        /**
-         * An element to display adjacent to the request button.
-         */
-        alternativeActionElement: PropTypes.element,
-    }
+        defaultMessage: string;
+    };
+
+    /**
+     * True if the {error} placeholder for the `errorMessage` property should include both
+     * the `message` and `detailed_error` properties of the error returned from the server,
+     * otherwise false to include only the `message` property.
+     */
+    includeDetailedError?: boolean;
+
+    /**
+     * An element to display adjacent to the request button.
+     */
+    alternativeActionElement?: React.ReactNode;
+};
+
+type State = {
+    busy: boolean;
+    fail: string | null;
+    success: boolean;
+}
 
-    static defaultProps = {
-        id: null,
+export default class RequestButton extends React.PureComponent<Props, State> {
+    static defaultProps: Partial<Props> = {
         disabled: false,
```

Comment:
since the `id` props was optional i remove the default `null` value for `id` because there will be an error if we pass null to `div` as it only accepts `string` or `undefined`